src/traversal.js

```diff
@@ -43,13 +57,17 @@ var exitNode = function(node) {
 
 
 /**
- * Marks a parent as having visited a child.
+ * Marks node's parent as having visited node.
  * @param {!Element} parent
  * @param {!Node} child
  */
-var markVisited = function(parent, child) {
-  var data = getData(parent);
-  data.lastVisitedChild = child;
+var markVisited = function(node) {
+  var walker = getWalker();
+  var parent = walker.getCurrentParent();
+  if (parent) {
```

Comment:
With the call from firstChild removed, I don't think there should ever be a case where we don't have a parent.